ckanext/dcatapchharvest/harvesters_i14y.py

```diff
@@ -0,0 +1,206 @@
+                {},
+                {'id': dataset_dict['owner_org']}
+            )
+            dataset_organization_name = dataset_organization['name']
+
+        except NotFound:
+            raise ValueError(
+                _('The selected organization was not found.')  # noqa
+            )
+
+        if dataset_dict.get('identifier'):
+            guid = dataset_dict['identifier']
+            try:
+                # check if the owner_org matches the identifier
+                if '@' in guid:
+                    org_name = guid.split('@')[-1]  # get last element
+                    org = model.Group.by_name(org_name)
+                    if not org:
+                        error_msg = (
+                            'The organization in the dataset identifier (%s) '
+                            'does not not exist. ' % org_name
+                        )
+                        log.error(error_msg)
+                        self._save_gather_error(error_msg, self.harvest_job)
+                        return None
+
+                    if org.id != dataset_dict['owner_org']:
+                        error_msg = (
+                            'The organization in the dataset identifier (%s) '
+                            'does not match the organization in the harvester '
+                            'config (%s)' % (org.id, dataset_dict['owner_org'])
+                        )
+                        log.error(error_msg)
+                        self._save_gather_error(error_msg, self.harvest_job)
+                        return None
+                else:
+                    # save original I14Y identifier and create a new identifier of the form <id>@<slug>,# noqa
+                    # where slug is an organization name
+                    dataset_dict['identifier_i14y'] = dataset_dict['identifier']  # noqa
+                    dataset_dict['identifier'] = \
+                        dataset_dict['identifier'] + '@' + dataset_organization_name  # noqa
+
+            except Exception as e:
+                log.exception("Error when getting identifier: %s" % e)
+                return None
+            return dataset_dict['identifier']
+
+        for extra in dataset_dict.get('extras', []):
+            if extra['key'] == 'uri' and extra['value']:
+                return extra['value']
+
+        if dataset_dict.get('uri'):
+            return dataset_dict['uri']
+
+        for extra in dataset_dict.get('extras', []):
+            if extra['key'] == 'identifier' and extra['value']:
+                return extra['value']
+
+        for extra in dataset_dict.get('extras', []):
+            if extra['key'] == 'dcat_identifier' and extra['value']:
+                return extra['value']
+
+        if dataset_dict.get('name'):
+            guid = dataset_dict['name']
+            if source_url:
+                guid = source_url.rstrip('/') + '/' + guid
+
+        return guid
+
+    def _gen_new_name(self, title):
```

Comment:
Do we need this method here if it's in the parent class and we are not changing it?